clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/RmdSerDe.java

```diff
@@ -48,7 +48,9 @@ public RmdSerDe(StringAnnotatedStoreSchemaCache annotatedStoreSchemaCache, int r
    * @return A {@link RmdWithValueSchemaId} object composed by extracting the value schema ID from the
    *    * header of the replication metadata.
    */
-  public RmdWithValueSchemaId deserializeValueSchemaIdPrependedRmdBytes(byte[] valueSchemaIdPrependedBytes) {
+  public void deserializeValueSchemaIdPrependedRmdBytes(
```

Comment:
Why change the signature here?
We are not trying to reuse the passed `RmdWithValueSchemaId`, are we?

Reply:
Not reused, but use it in this single call. This method is called by getReplicationMetadataAndSchemaId() in which it already is able to get the old rmd manifest. and this particular inner method also need to set rmd record and other info into RmdWithValueSchemaId class, so I passed it into this method.

Reply:
I see.

Can we update the javadoc to reflect the change? and also remove unused code?